### ISSUE TITLE
Error with `ggadjustedcurves()` : grouping factor has to be included in the coxph model

### ISSUE THREAD

**Issue by AlexisZr** [kassambara/survminer#623]
### Expected behavior 
As explicitly noted in the `survminer` cheatsheet:
> is not necessary to include the grouping factor in the Cox model
### Actual behavior
Despite following the exemplified information presented in the cheatsheet, the `ggadjustedcurves()` function exclusively generates a solitary survival curve without taking into account the provided grouping factor (which is not included in the cox model). To ascertain the veracity of the issue, I replicated the aforementioned example from the cheatsheet, which once more yielded solely one survival curve as opposed to the illustrated four.
### Steps to reproduce the problem
```
lung$sex <- ifelse(lung$sex == 1, 
                   "Male", "Female")
fit <- coxph(Surv(time, status) ~ 
               sex + ph.ecog + age +
               strata(sex), 
             data = lung)

lung$age3 <- cut(lung$age, 
                 c(35,55,65,85))
ggadjustedcurves(fit, data=lung, 
                 variable="age3")
```

**Comment by markhwhiteii** [comment]
+1 to this. The cheat sheet—and the README—both mention this is actually supposed to be done with the `ggcoxadjustedcurves` function, but that function is nowhere in the repository and thus not exported when downloading either from CRAN or GitHub.

**Comment by markhwhiteii** [comment]
@AlexisZr I found a workaround where you use the `predict` method:

```
# fit model
fit <- coxph(Surv(time, status) ~ sex + ph.ecog + age, data = lung)

# make prediction data
ages <- c(35, 55, 65, 85)
times <- min(lung$time):max(lung$time)
fit_pred <- data.frame(
  age = rep(ages, each = length(times)), 
  time = rep(times, length(ages)),
  sex = "Female", # holding this
  ph.ecog = 1, # and this constant
  status = 0 # doesnt matter if 1 or 0, you get the same prediction
)

# predict survival
fit_pred$pred <- predict(fit, fit_pred, type = "survival")

# plot it
ggplot(fit_pred, aes(x = time, y = pred)) +
  geom_line(aes(color = factor(age), group = factor(age)))
```

**Comment by kassambara** [comment]
Good catch — this is a **default-changed** gotcha, and the cheatsheet note is actually right *for the method it assumed*. Details:

`ggadjustedcurves()` has four methods: `"conditional"` (the current **default**), `"average"`, `"marginal"`, and `"single"`.

- The **default `"conditional"`** builds each group's curve by averaging the fitted model's predictions over the data. If the grouping variable is **not in the Cox model**, it doesn't enter the linear predictor, so every group's curve comes out **identical** — the "single curve" you saw (the 3 lines are drawn on top of each other).
- The cheatsheet's "not necessary to include the grouping factor" refers to the **marginal-style** adjustment. Use `method = "average"` (or `"marginal"`) and your original model works — distinct curves, grouping variable *not* in the model:

```r
library(survival); library(survminer)
lung$age3 <- cut(lung$age, c(35, 55, 65, 85))
lung2 <- lung[!is.na(lung$age3), ]
fit <- coxph(Surv(time, status) ~ sex + ph.ecog + age, data = lung2)   # age3 NOT in model

ggadjustedcurves(fit, data = lung2, variable = "age3", method = "average", palette = "jco")
```

![distinct adjusted curves with method = "average", variable not in model](https://i.imgur.com/YGMUzbX.png)

So you have two options: keep the default `"conditional"` **and put the grouping variable in the model**, or switch to `method = "average"`/`"marginal"` to reproduce the cheatsheet without adding it.

I'll leave this issue **open** for two documentation/robustness follow-ups:
1. The cheatsheet/README still reference **`ggcoxadjustedcurves()`**, which has been **removed** (as @markhwhiteii noted) — that reference should be updated to `ggadjustedcurves(..., method = "average")`, and the default-method change called out.
2. @markhwhiteii's `xtfrm`/`'>' not meaningful for factors` crash is a **separate** bug (the tibble/factor-column cluster, cf. #501/#628) that should be reproduced and fixed independently.

Thanks also to @azaras14 for the [`adjustedCurves`](https://github.com/RobinDenz1/adjustedCurves) pointer — a good alternative with several adjustment methods.

**Comment by kassambara** [comment]
Addressed in the development version (0.5.2.999) — merged via #747.

To make this less confusing, `ggadjustedcurves()`/`surv_adjustedcurves()` now **warn** when the default `method = "conditional"` is used with a grouping variable that is **not in the Cox model** — the exact case where every group's curve comes out identical:

```r
library(survival); library(survminer)
lung$age3 <- cut(lung$age, c(35, 55, 65, 85))
lung2 <- lung[!is.na(lung$age3), ]
fit <- coxph(Surv(time, status) ~ sex + ph.ecog + age, data = lung2)   # age3 NOT in model

ggadjustedcurves(fit, data = lung2, variable = "age3")                 # default = "conditional"
#> Warning: method = "conditional" (the default) builds each group's curve from the
#> Cox model, but the grouping variable 'age3' is not in the model, so the curves for
#> all groups are identical. Use method = "average" or "marginal" to draw distinct
#> curves for a variable that is not in the model, or add 'age3' to the Cox model.

ggadjustedcurves(fit, data = lung2, variable = "age3", method = "average")   # distinct curves
```

![conditional (identical curves) vs average (distinct curves)](https://i.imgur.com/SuIZDkR.png)

Left: the default `"conditional"` draws identical curves (the three age-group lines overlap) — now with a warning explaining why. Right: `method = "average"` gives the distinct curves you'd expect for a variable not in the model, matching the cheatsheet's intent. The warning is message-only — computed curves are unchanged — and does not fire for `"average"`/`"marginal"`/`"single"` or when the variable is in the model.

(Also fixed the README, which still referenced the removed `ggcoxadjustedcurves()` → `ggadjustedcurves()`.)

```r
remotes::install_github("kassambara/survminer")
```

Thanks @AlexisZr, @markhwhiteii, @azaras14 for the reports.
